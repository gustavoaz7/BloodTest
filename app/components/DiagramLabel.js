import React from 'react';
import { View, ViewPropTypes } from 'react-native';
import styled from 'styled-components/native';
import Text from './base/Text';
import { testResultType } from '../types';

export default function DiagramLabel({ style, result: { color, percentage, biomarker } }) {
  return (
    <Container style={style}>
      <ColoredDot color={color} />
      <Percentage>{`${percentage}%`}</Percentage>
      <Biomarker>{biomarker}</Biomarker>
    </Container>
  );
}

const Container = styled(View)`
  flex-direction: row;
  align-items: center;
`;

const ColoredDot = styled(View)`
  width: 10px;
  height: 10px;
  border-radius: 5px;
  ${({ color }) => `background-color: ${color}`};
`;

const Percentage = styled(Text)`
  margin: 0 6px;
`;

const Biomarker = styled(Text)``;

DiagramLabel.propTypes = {
  result: testResultType.isRequired,
  style: ViewPropTypes.style,
};

DiagramLabel.defaultProps = {
  style: null,
};
