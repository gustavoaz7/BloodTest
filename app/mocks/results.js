export const resultsMock = [
  {
    id: '0b82fed2-a091-4318-ba02-4d5e6192d6f7',
    name: 'Citlalli Kuhic',
    email: 'Berniece.Haley42@hotmail.com',
    samples: [
      {
        time: 'morning',
        value: 0.48,
      },
      {
        time: 'afternoon',
        value: 0.39,
      },
      {
        time: 'evening',
        value: 0.19,
      },
    ],
  },
  {
    id: 'acf65e6a-1878-4cd9-bf6f-440dafd4fa18',
    name: 'Rachael Lehner',
    email: 'Teagan_Ratke83@gmail.com',
    samples: [
      {
        time: 'morning',
        value: 0.67,
      },
      {
        time: 'afternoon',
        value: 0.31,
      },
      {
        time: 'evening',
        value: 0.24,
      },
    ],
  },
  {
    id: 'ca285be1-0d6b-42f6-bfec-b3aeaa2aaf8f',
    name: 'Willa Quitzon',
    email: 'Palma17@yahoo.com',
    samples: [
      {
        time: 'morning',
        value: 0.73,
      },
      {
        time: 'afternoon',
        value: 0.19,
      },
      {
        time: 'evening',
        value: 0.14,
      },
    ],
  },
  {
    id: '95eba822-a7b3-4938-8c94-005733701c1f',
    name: 'Miss Audie Moore',
    email: 'Giuseppe19@hotmail.com',
    samples: [
      {
        time: 'morning',
        value: 0.26,
      },
      {
        time: 'afternoon',
        value: 0.09,
      },
      {
        time: 'evening',
        value: 0.29,
      },
    ],
  },
  {
    id: '4fea89aa-bcf7-42b0-a5e8-202fc03a02fe',
    name: 'Yesenia Nicolas',
    email: 'Maeve_Jacobs14@yahoo.com',
    samples: [
      {
        time: 'morning',
        value: 0.78,
      },
      {
        time: 'afternoon',
        value: 0.26,
      },
      {
        time: 'evening',
        value: 0.24,
      },
    ],
  },
  {
    id: '17d11f05-2580-49b5-bb17-b786209c4792',
    name: 'Ms. Harmon Kunze',
    email: 'Johnson90@yahoo.com',
    samples: [
      {
        time: 'morning',
        value: 0.47,
      },
      {
        time: 'afternoon',
        value: 0.09,
      },
      {
        time: 'evening',
        value: 0.09,
      },
    ],
  },
];
