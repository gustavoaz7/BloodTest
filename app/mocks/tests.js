export const testsMock = [
  {
    id: '5d41b693ed44f7ee9f9436d0',
    title: 'Healthcare',
    date: '17.07.2019',
    result: [
      {
        biomarker: 'fats',
        color: '#e30613',
        percentage: 25,
      },
      {
        biomarker: 'carbohydrates',
        color: '#D3235D',
        percentage: 25,
      },
      {
        biomarker: 'vitamins',
        color: '#547F81',
        percentage: 25,
      },
      {
        biomarker: 'magnesium',
        color: '#0085FF',
        percentage: 25,
      },
    ],
  },
  {
    id: '5d41b6938f1360cdde5ecc89',
    title: 'Advanced',
    date: '08.07.2019',
    result: [
      {
        biomarker: 'protein',
        color: '#E2D669',
        percentage: 25,
      },
      {
        biomarker: 'vitamins',
        color: '#ffcc00',
        percentage: 25,
      },
      {
        biomarker: 'magnesium',
        color: '#e30613',
        percentage: 25,
      },
      {
        biomarker: 'carbohydrates',
        color: '#0085FF',
        percentage: 25,
      },
    ],
  },
  {
    id: '5d41b693084f4f55217fd864',
    title: 'Sport',
    date: '30.07.2019',
    result: [
      {
        biomarker: 'fiber',
        color: '#E2D669',
        percentage: 25,
      },
      {
        biomarker: 'fats',
        color: '#D3235D',
        percentage: 25,
      },
      {
        biomarker: 'magnesium',
        color: '#547F81',
        percentage: 25,
      },
      {
        biomarker: 'vitamins',
        color: '#0085FF',
        percentage: 25,
      },
    ],
  },
];
