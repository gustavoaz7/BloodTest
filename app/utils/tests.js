export const flushPromises = () => new Promise(setImmediate);

export const stackNavigationProps = {
  state: {},
  dispatch() {},
  goBack() {},
  dismiss() {},
  navigate() {},
  getParam() {},
  setParams() {},
  emit() {},
  addListener() {},
  isFocused() {},
  isFirstRouteInParent() {},
  dangerouslyGetParent() {},
  router() {},
};
