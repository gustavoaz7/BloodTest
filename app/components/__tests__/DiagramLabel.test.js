import React from 'react';
import { shallow } from 'enzyme';
import DiagramLabel from '../DiagramLabel';
import { testsMock } from '../../mocks/tests';

describe('DiagramLabel', () => {
  const result = testsMock[0].result[0];
  it('renders without crashing', () => {
    shallow(<DiagramLabel result={result} />);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<DiagramLabel result={result} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('accepts custom style', () => {
    const style = { backgroundColor: 'black' };
    const wrapper = shallow(<DiagramLabel style={style} result={result} />);

    expect(wrapper.props().style).toEqual(style);
  });
});
