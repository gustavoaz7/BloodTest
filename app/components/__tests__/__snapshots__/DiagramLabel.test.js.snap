// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiagramLabel matches snapshot 1`] = `
<Styled(View)
  style={null}
>
  <Styled(View)
    color="#e30613"
  />
  <Styled(Styled(Text))>
    25%
  </Styled(Styled(Text))>
  <Styled(Styled(Text))>
    fats
  </Styled(Styled(Text))>
</Styled(View)>
`;
